test/integration/mutations/request.test.js

```diff
@@ -0,0 +1,355 @@
+var expect = require('chai').expect;
+
+describe('request mutations', function () {
```

Comment:
All of these blocks need to have a test that checks for the pre-mutated state being valid.